### PR TITLE
Allow custom redirect for landing page

### PR DESCRIPTION
**Pull request by techknowlogick**
Allow for a wider variety of landing page redirects

### CODE REVIEW

File: modules/setting/setting.go [hunk 1]
```diff
@@ -776,15 +777,19 @@ func loadFromConf(allowEmpty bool, extraConfig string) {
...
 	case "organizations":
 		LandingPageURL = LandingPageOrganizations
 	case "login":
 		LandingPageURL = LandingPageLogin
-	default:
-		LandingPageURL = LandingPageHome
+	case "custom":
+		customUrl := LandingPage(sec.Key("LANDING_PAGE_CUSTOM").MustString(""))
+		if customUrl != "" {
+			LandingPageURL = customUrl
```
Comment (delvh) [inline modules/setting/setting.go]:
```suggestion
		customUrl := sec.Key("LANDING_PAGE_CUSTOM").MustString("")
		if customUrl != "" {
			LandingPageURL = LandingPage(customUrl)
```
(slightly more performant in an edge case even though it probably took longer to write compared to the time it saves...)

File: custom/conf/app.example.ini [hunk 2]
```diff
@@ -237,10 +237,13 @@ RUN_MODE = ; prod
 ;; PPROF_DATA_PATH, use an absolute path when you start gitea as service
 ;PPROF_DATA_PATH = data/tmp/pprof
 ;;
-;; Landing page, can be "home", "explore", "organizations" or "login"
+;; Landing page, can be "home", "explore", "organizations", "login", or "custom"
 ;; The "login" choice is not a security measure but just a UI flow change, use REQUIRE_SIGNIN_VIEW to force users to log in.
 ;LANDING_PAGE = home
```
Comment (delvh) [inline custom/conf/app.example.ini:242]: That is one approach.
Another would have been to automatically use an unrecognized value as the landing page URL.
I'm fine with both approaches.

**Review by delvh — APPROVED** [review]

File: custom/conf/app.example.ini (hunk 2 above)
Comment (wxiaoguang) [reply custom/conf/app.example.ini:242]: We should avoid introducing more setting options here. Indeed we can reuse the `LANDING_PAGE`

`LANDING_PAGE=home`:  builtin home URL
`LANDING_PAGE=/org/repo`: the `/org/repo` URL if the value starts with a slash.


```- `LANDING_PAGE_CUSTOM`: ****: Landing page for unauthenticated users, \[home, explore, organizations, login\], or an absolute URL path like "/user/profile" ```

Comment (lunny) [reply custom/conf/app.example.ini:242]: Or `LANDING_PAGE= https://anotherwebsite`

Comment (techknowlogick) [reply custom/conf/app.example.ini:242]: I've updated it to just be the one option now that covers all cases as discussed above

**Review by a1012112796 — APPROVED** [review]

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 3]
```diff
@@ -300,8 +300,7 @@ The following configuration set `Content-Type: application/vnd.android.package-a
 - `ENABLE_GZIP`: **false**: Enable gzip compression for runtime-generated content, static resources excluded.
 - `ENABLE_PPROF`: **false**: Application profiling (memory and cpu). For "web" command it listens on localhost:6060. For "serv" command it dumps to disk at `PPROF_DATA_PATH` as `(cpuprofile|memprofile)_<username>_<temporary id>`
 - `PPROF_DATA_PATH`: **data/tmp/pprof**: `PPROF_DATA_PATH`, use an absolute path when you start Gitea as service
-- `LANDING_PAGE`: **home**: Landing page for unauthenticated users \[home, explore, organizations, login\].
-
+- `LANDING_PAGE`: **home**: Landing page for unauthenticated users \[home, explore, organizations, login, **custom**\]. Where custom would instead be any URL such as "/org/repo" or even "https://anotherwebsite.com"
```
Comment (wxiaoguang) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md:303]: image

The URL in document becomes a link on the document page, it seems not good.

Comment (techknowlogick) [reply docs/content/doc/advanced/config-cheat-sheet.en-us.md:303]: Hmm.... perhaps if I wrap it in ` that'll escape the link

Comment (jolheiser) [reply docs/content/doc/advanced/config-cheat-sheet.en-us.md:303]: I vaguely recall a GH action for netlify previews, maybe that would be nice for PRs touching docs.
(Until we move to gitea.com, but we could find an alternative then)

**Pull request merged**